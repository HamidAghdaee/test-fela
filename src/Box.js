import React from "react";
import { createComponent } from "react-fela";

const boxRules = ({ size = 10 }) => ({
  width: size + "px",
  height: size + "px",
  backgroundColor: "red"
});

const Box = createComponent(boxRules);
export default Box;
