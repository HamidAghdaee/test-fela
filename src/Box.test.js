import React from "react";
import PropTypes from "prop-types";
import Box from "./Box";
import shallow from "./felaShallow";

it("renders without crashing", () => {
  const container = shallow(<Box>hello</Box>);
  expect(container.snapshot().component).toMatchSnapshot();
  expect(container.snapshot().styles).toMatchSnapshot();
});
