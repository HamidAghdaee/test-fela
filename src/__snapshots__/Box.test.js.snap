// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<boxRules
  _felaTheme={Object {}}
>
  hello
</boxRules>
`;

exports[`renders without crashing 2`] = `""`;
